File: ios/Plugin/socketIOPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(socketIOPlugin, "socketIO",
           CAP_PLUGIN_METHOD(connect, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(emit, CAPPluginReturnNone);
           )

File: ios/Plugin/socketIOPlugin.swift
import Foundation
import Capacitor
import SocketIO
/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(socketIOPlugin)
public class socketIOPlugin: CAPPlugin {
    private let implementation = socketIO()
    private var server = "http://localhost";
    private var socket: SocketIOClient? = nil;
    private var manager: SocketManager? = nil;

    @objc func connect(_ call: CAPPluginCall) {
      server=call.getString("server") ?? "sample";
      manager = SocketManager(socketURL: URL(string:server)!);
        socket=manager?.socket(forNamespace: "/");
        socket?.connect();
        call.resolve(["socket":socket!]);
    }
    @objc func emit(_ call: CAPPluginCall) {
        let key = call.getString("key") ?? "key";
        let data:Any = call.getObject("data") ?? "data";
        socket!.emit(key, data as! [Any]);
        call.resolve();
    }
}
